### PR TITLE
Issue 25156 creating new url parameter metric

### PR DESCRIPTION
**Pull request by freddyucv**
We need to create a new Metric how it is explained here https://github.com/dotCMS/core/issues/25156

### Proposed Changes

* I am including a psotman test to set a URL Parameter Goal to an Experiment

https://github.com/dotCMS/core/pull/25534/files#diff-039db8b104840e0437baac989d496b376f2d496bc9fc89bb79046e38e9d563d3R3

* Create a new Exists Operator

https://github.com/dotCMS/core/pull/25534/files#diff-75510bc7dededdfed8b986435f8be1f23e53e726a1f3e3e79e9b2b48e8930cd6R61

* We need a new Parameter to set the Experiment Page and later check if this page was visited before the QueryParameter was set

* We need to skip some Parameter to be check so we are going to create a new validate attribute in the Parameter class, we don't need that the visitBefore to be check on a different place 

https://github.com/dotCMS/core/pull/25534/files#diff-75510bc7dededdfed8b986435f8be1f23e53e726a1f3e3e79e9b2b48e8930cd6R92

https://github.com/dotCMS/core/pull/25534/files#diff-fa584a104c84bee14f1387bb8395d0a4989a51f282912cc7183d96f2e4fff42aR48

* Create the new Metric Type 

https://github.com/dotCMS/core/pull/25534/files#diff-a99d90c3cfd57cac21364804699421782fdb91332fccc3b56ad0441b375b7ba4R35

* Create a new Getter to get the Query Parameters from the Event's Url

https://github.com/dotCMS/core/pull/25534/files#diff-36b5300d70021c5b40e8f95608c2508decb9ffee5d492464a77b625a90ca1518R14

* Create a new Transformer to transform the QueryParameter to String and then compare it on the Operators

https://github.com/dotCMS/core/pull/25534/files#diff-fe9e0a514677658c9fa891a154de15c299c852ab554e3e50affba9aefef78564R14

* Automatically set the value of the visitBefore Condition when an Experiment with Url Parameter Goal is saved

https://github.com/dotCMS/core/pull/25534/files#diff-5aa71567ff53ee229d25c2386cce3169c657f37a3e04cf27b75b3fd029537a7cR236

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/QueryParameter.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.dotcms.analytics.metrics;
+
+/**
+ * Represents a Url Query Parameter
+ */
+public class QueryParameter {
+    private String name;
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/analytics/metrics/QueryParameter.java:7]: Can these fields be `final`?

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/QueryParameterValuesGetter.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+            return getQueryParams(query).entrySet().stream()
+                    .map(entry -> new QueryParameter(entry.getKey(), entry.getValue()))
+                    .collect(Collectors.toList());
+        } catch (URISyntaxException e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private static Map<String, String> getQueryParams(String query) {
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/analytics/metrics/QueryParameterValuesGetter.java:37]: I wonder is there's already a library that takes care of the parsing of a http query.
If not, shouldn't these methods exist in "Utils" class?

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/QueryParameterValuesGetter.java (hunk 2 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/analytics/metrics/QueryParameterValuesGetter.java:37]: I think the `om.dotcms.util.HttpRequestDataUtil` has the method Freddy needs.

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/analytics/metrics/QueryParameterValuesGetter.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/analytics/metrics/QueryParameterValuesGetter.java]: done https://github.com/dotCMS/core/pull/25534/commits/604cb4212cb98ec878314c443a29eddd27185e21#diff-0be3d9cc6974385beca6556e8697c31c3a235292a46c189d54017ae50c2fdf39R7

the method that exists is this one 

https://github.com/dotCMS/core/blob/604cb4212cb98ec878314c443a29eddd27185e21/dotCMS/src/main/java/com/dotcms/util/HttpRequestDataUtil.java#L153

but it is not exactly what I need

**Review by jcastro-dotcms — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**